### PR TITLE
Improvement : Emit events

### PR DESCRIPTION
**Pull request by justinehell**
- Use of leto-modelizer-plugin-core #0.15.1
- Update `TerraformDrawer.js` and `TerraformPlugin.js`
- Add events in `TerraformParser.js` and `TerraformRenderer.js`

### CODE REVIEW

File: src/parser/TerraformParser.js [hunk 1]
```diff
@@ -24,13 +24,44 @@ class TerraformParser extends DefaultParser {
...
    */
-  parse(inputs = []) {
+  parse(inputs = [], parentEventId = null) {
     this.pluginData.components = [];
+
     const listener = new TerraformListener(this.pluginData.definitions.components);
+
     inputs
-      .filter(({ content }) => content !== null)
+      .filter(({ content }) => {
```
Comment (Zorin95670) [inline src/parser/TerraformParser.js:35]: `{ content, path }`

File: src/parser/TerraformParser.js [hunk 2]
```diff
@@ -24,13 +24,44 @@ class TerraformParser extends DefaultParser {
...
+        if (content !== null || content?.trim() === '') {
+          return true;
+        }
+
+        this.pluginData.emitEvent({
+          parent: parentEventId,
+          type: 'Parser',
+          action: 'read',
+          status: 'warning',
+          data: {
```
Comment (Zorin95670) [inline src/parser/TerraformParser.js:45]: + `files: [path],`

File: src/parser/TerraformParser.js [hunk 3]
```diff
@@ -24,13 +24,44 @@ class TerraformParser extends DefaultParser {
...
-  parse(inputs = []) {
+  parse(inputs = [], parentEventId = null) {
     this.pluginData.components = [];
+
     const listener = new TerraformListener(this.pluginData.definitions.components);
+
     inputs
-      .filter(({ content }) => content !== null)
+      .filter(({ content }) => {
+        if (content !== null || content?.trim() === '') {
```
Comment (Zorin95670) [inline src/parser/TerraformParser.js:36]: `if (content && content.trim() !== '') {`

File: src/parser/TerraformParser.js (hunk 2 above)
Comment (justinehell) [reply src/parser/TerraformParser.js:46]: done

File: src/models/TerraformPlugin.js [hunk 4]
```diff
@@ -14,13 +14,19 @@ import packageInfo from 'package.json';
 class TerraformPlugin extends DefaultPlugin {
   /**
    * Default constructor.
+   *
+   * @param {object} [props={}] - Object that contains all properties to set.
```
Comment (k-y0u) [inline src/models/TerraformPlugin.js:18]: shouldn't it be `{Object}` ? like `{Function}` under...

File: src/parser/TerraformParser.js [hunk 5]
```diff
@@ -24,13 +24,45 @@ class TerraformParser extends DefaultParser {
    * Convert the content of files into Components.
    *
    * @param {FileInput[]} [inputs=[]] - Data you want to parse.
+   * @param {string} [parentEventId=null] - Parent event id.
```
Comment (k-y0u) [inline src/parser/TerraformParser.js:27]: shouldn't it be `{String}`, like `{FileInput[]}` above ?

File: src/render/TerraformRenderer.js [hunk 6]
```diff
@@ -35,9 +35,10 @@ class TerraformRenderer extends DefaultRender {
   /**
    * Convert all provided components and links in terraform files.
    *
+   * @param {string} [parentEventId=null] - Parent event id.
```
Comment (k-y0u) [inline src/render/TerraformRenderer.js:38]: same

File: src/render/TerraformRenderer.js [hunk 7]
```diff
@@ -50,23 +51,40 @@ class TerraformRenderer extends DefaultRender {
...
       ),
+      parentEventId,
     );
   }
 
   /**
    * Render files from related components.
    *
    * @param {Map<string,Component>} map - Component mapped by file name.
+   * @param {string} [parentEventId=null] - Parent event id.
```
Comment (k-y0u) [inline src/render/TerraformRenderer.js:62]: same

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/parser/TerraformParser.js (hunk 5 above)
Comment (Zorin95670) [reply src/parser/TerraformParser.js:27]: no in jsdoc is string :)

**Review by Zorin95670 — DISMISSED** [review]

File: src/models/TerraformPlugin.js (hunk 4 above)
Comment (Zorin95670) [reply src/models/TerraformPlugin.js:18]: It's manage by linter and jsdoc, if you set Object we have a linter error. So it's good like that !

File: src/render/TerraformRenderer.js (hunk 6 above)
Comment (Zorin95670) [reply src/render/TerraformRenderer.js:38]: It's manage by linter and jsdoc, if you set String we have a linter error. So it's good like that !

File: src/render/TerraformRenderer.js (hunk 7 above)
Comment (Zorin95670) [reply src/render/TerraformRenderer.js:62]: It's manage by linter and jsdoc, if you set String we have a linter error. So it's good like that !

**Review by k-y0u — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**